core/commands/publish.go

```diff
@@ -89,7 +91,8 @@ Publish an <ipfs-path> to another public key (not implemented):
 		}
 
 		// TODO n.Keychain.Get(name).PrivKey
-		output, err := publish(n, n.PrivateKey, p)
+		// TODO(cryptix): is req.Context().Context a child of n.Context()?
```

Comment:
ctxgroup is old and should be removed. ctxgroup should be replaced by goprocess. the only reason i haven't done it is that goprocess isn't perfect yet-- i believe it has a memleak atm. :( 

if we want to abandon goprocess and ctxgroup for something even simpler, down to think about it. but the problems are pretty annoying -- many cancellations, want semantics where you know the async op has completed, want to query (without blocking) whether something is closed or not, want to wait on things to finish, etc. This problem is not perfectly solved by anything out there-- ctxgroup -> goprocess is _an_ attempt at it. (I know there is a similar internal package at google).